modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleServiceIT.java

```diff
@@ -824,6 +831,96 @@ public void testReenableDataStreamLifecycle() throws Exception {
         });
     }
 
+    public void testLifecycleAppliedToFailureStore() throws Exception {
+        // We configure a lifecycle with downsampling to ensure it doesn't fail
+        DataStreamLifecycle lifecycle = DataStreamLifecycle.newBuilder()
+            .dataRetention(2_000)
+            .downsampling(
+                new DataStreamLifecycle.Downsampling(
+                    List.of(
+                        new DataStreamLifecycle.Downsampling.Round(
+                            TimeValue.timeValueMillis(10),
+                            new DownsampleConfig(new DateHistogramInterval("10m"))
+                        )
+                    )
+                )
+            )
+            .build();
+
+        putComposableIndexTemplate("id1", """
+            {
+              "properties": {
+                 "@timestamp": {
+                   "type": "date",
+                   "format": "epoch_millis"
+                 },
+                 "flag": {
+                   "type": "boolean"
+                 }
+             }
+            }""", List.of("metrics-fs*"), Settings.builder().put("index.number_of_replicas", 0).build(), null, lifecycle, true);
+
+        String dataStreamName = "metrics-fs";
+        CreateDataStreamAction.Request createDataStreamRequest = new CreateDataStreamAction.Request(dataStreamName);
+        client().execute(CreateDataStreamAction.INSTANCE, createDataStreamRequest).get();
+
+        indexInvalidFlagDocs(dataStreamName, 1);
+
+        // Let's verify the rollover
+        assertBusy(() -> {
+            GetDataStreamAction.Request getDataStreamRequest = new GetDataStreamAction.Request(new String[] { dataStreamName });
```

Comment:
Thinking out loud, and this probably also holds for the existing tests, isn't it theoretically possible for this test to fail if this GET request takes longer than normal to return and in the meantime DSL has executed 3 times (once every second), causing the old write index to already have been deleted? I don't immediately have a solution for this, and I think the chances are pretty low, so this is more a question to see if I understand correctly.

Reply:
Yeah that should work. That does mean that we're changing the test a bit as we're _updating_ a lifecycle halfway through, but I guess that isn't too big of a deal. An alternative would be to simply increase the retention, but that would _always_ slow down the test, so not ideal either. Let's do your suggestion :+1:.